### PR TITLE
chore: add warning deprecation of --no-workspace-strict-match

### PR DESCRIPTION
**Pull request by ghiscoding**
## Types of changes

- [ ] Chore (change that has absolutely no effect on users)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/publish/src/__tests__/publish-command.spec.ts [hunk 1]
```diff
@@ -483,6 +484,19 @@ Map {
...
 
+  describe('--no-workspace-strict-match', () => {
+    it('shows warning that this is the default behavior and that this option is no longer needed', async () => {
+      const cwd = await initFixture('normal');
+
+      await lernaPublish(cwd)('--no-workspace-strict-match');
+
+      const logMessages = loggingOutput('warn');
+      expect(logMessages).toContain(
+        'Providing --no-workspace-strict-match is deprecated and will be removed in future version, we will make "workspace:" protocol strict matching in every case.'
```
Comment (meteorlxy) [inline packages/publish/src/__tests__/publish-command.spec.ts:495]: I'm seeing this warning even if I didn't set this flag. Is this as expected?

File: packages/publish/src/publish-command.ts [hunk 2]
```diff
@@ -141,6 +141,13 @@ export class PublishCommand extends Command<PublishCommandOption> {
       );
     }
 
+    if (!this.options.workspaceStrictMatch) {
+      this.logger.warn(
+        'deprecation',
+        'Providing --no-workspace-strict-match is deprecated and will be removed in future version, we will make "workspace:" protocol strict matching in every case.'
```
Comment (meteorlxy) [inline packages/publish/src/publish-command.ts:147]: I'm seeing this warning even if I didn't set this flag. Is this as expected?

Comment (ghiscoding) [reply packages/publish/src/publish-command.ts:147]: @meteorlxy  oh really, no it's not expected at all, could you provide the options to help finding where that comes from?

Comment (meteorlxy) [reply packages/publish/src/publish-command.ts:147]: Uh, I was going to provide a repo and then noticed that you've fixed it. Thanks 👍

Comment (ghiscoding) [reply packages/publish/src/publish-command.ts:147]: yeah I just pushed a new release [v1.9.1](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.9.1), thanks for raising the problem, it should be working as expected now, let me know if you still see it. Thanks

**Pull request merged**